src/service/resources-impl.js

```diff
@@ -1721,7 +1729,17 @@ export class Resource {
    * @return {boolean}
    */
   renderOutsideViewport() {
-    return this.element.renderOutsideViewport();
+    const renders = this.element.renderOutsideViewport();
+    // Boolean interface, element is either always allowed or never allowed to
+    // render outside viewport.
+    if (renders === true || renders === false) {
+      return renders;
+    }
+    // Numeric interface, element is allowed to render outside viewport when it
+    // is within X times the viewport height of the current viewport.
+    const viewportBox = this.resources_.getViewport().getRect();
+    const distanceFromViewport = this.layoutBox_.top - viewportBox.bottom;
```

Comment:
I'd expect some negative logic here, no? It looks like this value can be negative a lot. So it should either involve abs or two different formulas for above/below viewport.